src/shared_types.rs

```diff
@@ -109,22 +138,17 @@ struct RawGuideline {
     identifier: Option<Identifier>,
 }
 
-impl Identifier {
-    fn is_valid(&self) -> bool {
-        self.0.len() <= 100 && self.0.bytes().all(|b| (0x20..=0x7E).contains(&b))
-    }
+fn is_valid_identifier(s: &str) -> bool {
+    s.len() <= 100 && s.bytes().all(|b| (0x20..=0x7E).contains(&b))
 }
 
 impl Serialize for Identifier {
     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
     where
         S: Serializer,
     {
-        if self.is_valid() {
-            serializer.serialize_str(&self.0)
-        } else {
-            Err(ser::Error::custom("Identifier must be at most 100 characters long and contain only ASCII characters in the range 0x20 to 0x7E."))
-        }
+        debug_assert!(is_valid_identifier(&self.0));
```

Comment:
Not a `ser::Error`?

Reply:
Because we now validate on construction, it should actually be impossible to have an invalid identifier; this is just a sanity check to that effect. I should have added a comment though. :)

Reply:
Ah, right.